### PR TITLE
fix(mock): enable arrayItems option for MSW-only mock configs (#3576)

### LINKED ISSUE #3576: arrayItems mock option is unavailable for MSW-only configs
## Problem

`arrayItems` is typed and implemented as a **faker-only** mock option. With an MSW-only config, you cannot opt into reusable array-item mock factories:

```ts
mock: {
  generators: [{ type: 'msw', arrayItems: true }],
},
```

TypeScript rejects `arrayItems` on the MSW generator entry, and generation ignores it even if forced.

## Expected

`arrayItems` should work for MSW-only setups. MSW already builds response mock data through the same faker mock pipeline (`getMockScalar` → `extractArrayItemMock`), so the feature is useful without also enabling a separate `faker` generator.

## Actual

- `arrayItems` lives on `FakerMockOptions`, not shared mock options.
- `shouldExtractArrayItemFactories` only checks `type: 'faker'` generator entries.

## Impact

Projects that generate only `.msw.ts` mocks cannot get exported item factories such as `getTenantResponseModelDtoMock` for `value: TenantResponseModelDto[]` in paginated responses.

### PR DESCRIPTION
**Pull request by kfrederix**
## Summary
- Move `arrayItems` from `FakerMockOptions` to `CommonMockOptions` so it is valid on MSW generator entries.
- Update `shouldExtractArrayItemFactories` to honor `arrayItems: true` on any mock generator, not only `type: 'faker'`.
- Add unit tests, an `mswArrayItems` snapshot fixture, and docs updates for both the Faker and MSW guides.
Fixes #3576.
## Test plan
- [ ] `vp run -F '@orval/mock' test`
- [ ] `vp run -F orval-tests generate:mock`
- [ ] `vp run -F orval-tests test:snapshots`
- [ ] Confirm MSW-only config `{ type: 'msw', arrayItems: true }` emits item factories (e.g. `getTenantResponseModelDtoMock`) in generated mock output

## Summary by CodeRabbit

* **New Features**
  * Added `arrayItems` option for Faker and MSW mock generators to emit reusable factory helpers for object-like array items in responses.

* **Documentation**
  * Updated Faker guide and reorganized Options so `arrayItems` is a shared mock option.
  * Added an MSW "Array Item Factories" section with examples and config snippets.

* **Tests**
  * Added tests, snapshots, sample OpenAPI spec and sample mock config demonstrating `arrayItems: true`.

### CODE REVIEW

**Comment by Hypenate** [comment]
@wadakatu this is another issue we observed on our own project.

**Comment by kfrederix** [comment]
Sorry for the sloppy commit history. I struggled a bit with the bun install on my machine so couldn't run any checks locally. But finally got it working now.

If you prefer to abandon this PR and re-open a clean one: just say so and I will do it.

**Comment by melloware** [comment]
@kfrederix no its fine!  Thanks for the PR!

**Comment by melloware** [comment]
@kfrederix sorry merge conflicts after the latest PR merge.

**Comment by kfrederix** [comment]
No worries. I'll resolve them and push an update soon.

**Comment by kfrederix** [comment]
@melloware @wadakatu the conflicts should be resolved now.

**Review by melloware — APPROVED** [review]
This LGTM to me. So just waiting on @wadakatu to review

**Review by wadakatu — APPROVED** [review]
**LGTM!** 👍

The core change is minimal and lands in the right place: `arrayItems` moves to `CommonMockOptions`, and `shouldExtractArrayItemFactories` relaxes from faker-only `find` to `.some()` over all non-function generator entries. Since the file-level dedup buckets are already keyed per generator type (`${scope}:${mockType}`), each pass stays self-contained — no duplicate declarations or dangling references.

**Verified locally** (on top of the green CI):
- `@orval/mock` unit tests: 236 passed
- Regenerated the `mswArrayItems` fixture — byte-identical to the committed snapshot, and `tsc --strict` passes on the output
- `vp fmt --check` passes

**Edge cases I exercised with temporary configs (all generated + type-checked clean):**
- **MSW-only over the full `faker-array-items` spec** (multiple operations sharing the same `$ref` item, inline items, `oneOf` items, nullable items, name-collision cases) — single deduped factory referenced by all 3 ops; inline/oneOf extracted, nullable/collide intentionally inlined, exactly matching the faker generator's behavior
- **`msw` + `faker` combined, single mode** — no duplicate declarations (`collapseInlineMockOutputs` drops the faker chunk as designed)
- **`faker { schemas: true }` + `msw { arrayItems: true }`** — the MSW output *imports* the consolidated factory from `model/index.faker.ts` instead of re-declaring it. Nice delegation
- **MSW-only, tags-split mode** — each tag file gets its own self-contained factory copy
- **split mode / custom faker `path`** with `arrayItems` only on the msw entry — both files compile, self-contained

Two non-blocking notes:

1. **Docs nit:** in `faker.mdx`, the Array Item Factories example config was switched from `type: 'faker'` to `type: 'msw'`. Since the MSW guide now carries its own msw example, keeping `'faker'` in the faker guide might be less surprising for readers copying from that page.
2. **Semantics observation (no action needed):** the gate is global across generator entries, so `[{ type: 'msw', arrayItems: true }, { type: 'faker' }]` also emits factories in the faker output even though that entry didn't opt in. This is symmetric with the pre-existing behavior (a faker opt-in already leaked into MSW output on master) and the docs correctly say "on any mock generator entry", so I think it's fine as-is. If per-entry control is ever wanted later, `context.activeMockOutputType` is already available to scope the gate.

File: packages/mock/src/msw/index.test.ts [hunk 1]
```diff
@@ -1399,6 +1399,142 @@ describe('generateMSW', () => {
...
+      },
+    },
+    output: {
+      target: 'test',
+      namingConvention: 'camelCase',
+      fileExtension: '.ts',
+      mode: 'single',
+      mock: {
+        generators: [{ type: OutputMockType.MSW, arrayItems: true, delay: false }],
+      },
```
[the remark this answers is not in the document]
Comment (wadakatu) [reply packages/mock/src/msw/index.test.ts]: Not reproducible: `vp fmt --check` passes locally on this branch (366 files, no issues) and the CI format gate is green on both ubuntu and windows. The generators literal in `msw/index.test.ts` matches the project formatting as-is.
